routers/private/hook_pre_receive.go

```diff
@@ -450,24 +444,42 @@ func generateGitEnv(opts *private.HookOptions) (env []string) {
 	return env
 }
 
-func loadUserAndPermission(ctx *gitea_context.PrivateContext, id int64) (user *user_model.User, perm models.Permission) {
-	user, err := user_model.GetUserByID(id)
+func (ctx *preReceiveContext) loadPusherAndPermission() {
+	if ctx.loadedPusher {
+		return
+	}
+
+	user, err := user_model.GetUserByID(ctx.opts.UserID)
 	if err != nil {
-		log.Error("Unable to get User id %d Error: %v", id, err)
+		log.Error("Unable to get User id %d Error: %v", ctx.opts.UserID, err)
 		ctx.JSON(http.StatusInternalServerError, private.Response{
-			Err: fmt.Sprintf("Unable to get User id %d Error: %v", id, err),
+			Err: fmt.Sprintf("Unable to get User id %d Error: %v", ctx.opts.UserID, err),
 		})
 		return
```

Comment:
Should it return a error, so code wouldn't assume that `user` and userPerm` is non-nill?

Reply:
Returning errors doesn't make code more simple at the moment.

The existing (old) logic is using `ctx.Written` to check errors in `loadPusherAndPermission`, the code in this PR just follows that and can guarantee that there is no nil access.

If we want to rewrite the error handling logic, there would be a bigger refactoring, which makes the PR more complex and harder to be approved.

Reply:
Hmm, I think I can make loadPusherAndPermission to return `bool`, then the logic doesn't change much, and can be more clear.

Reply:
improved, pushed.

Reply:
Ack